plugin-api/src/main/java/io/opentelemetry/sdk/resources/EaseAgentResource.java

```diff
@@ -0,0 +1,83 @@
+package io.opentelemetry.sdk.resources;
+
+import com.megaease.easeagent.plugin.api.config.ChangeItem;
+import com.megaease.easeagent.plugin.api.config.ConfigChangeListener;
+import com.megaease.easeagent.plugin.api.otlp.common.AgentAttributes;
+import com.megaease.easeagent.plugin.api.otlp.common.SemanticKey;
+import com.megaease.easeagent.plugin.bridge.EaseAgent;
+import io.opentelemetry.api.common.Attributes;
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAMESPACE;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class EaseAgentResource extends Resource implements ConfigChangeListener {
+    static volatile EaseAgentResource agentResource = null;
+
+    public EaseAgentResource() {
```

Comment:
Singleton pattern, the constructor shouldn't be accessible

Reply:
OK